labtool2.0/src/reducers/codeReviewReducer.js

```diff
@@ -18,12 +18,63 @@ const INITIAL_STATE = {
   filterActive: false
 }
 
-function shuffleArray(array) {
+/**
+ * Check if there is a repeated reviewer-toReview pair or the reviewer is same as toReview when code review round>1
+ */
+const checkRepeated = (currentSelections, reviewers, toReviews, round) => {
+  if (round === 'create') {
+    round = Object.keys(currentSelections).length
+  }
+  for (let i = 1; i < round; i++) {
+    for (let j = 0; j < reviewers.length; j++) {
+      if (currentSelections[i][reviewers[j]] === toReviews[j]) {
+        //repeated reviewer-toReview pair
+        return true
+      }
+      if (reviewers[j] === toReviews[j]) {
+        //reviewer is same as toReview
+        return true
+      }
+    }
+  }
```

Comment:
Maybe there would be a better way to check this? I think this might be quite slow if the course has hundreds of students?